python/src/scipp/plot/slicer.py

```diff
@@ -262,48 +273,44 @@ def axis_label_and_ticks(self, axis, data_array, name):
+        #     dim = data_array.labels[axis].dims[-1]
+        #     lab = axis
+        #     var = data_array.labels[lab]
+        # else:
+        dim = axis
+        # Convert to Dim object?
+        if isinstance(dim, str):
+            dim = Dim(dim)
+        # lab = None
+        make_fake_coord = False
+        fake_unit = None
+        if not data_array.coords.__contains__(dim):
+            make_fake_coord = True
+        else:
+            tp = data_array.coords[dim].dtype
+            if tp == dtype.vector_3_float64:
                 make_fake_coord = True
-            else:
-                tp = data_array.coords[dim].dtype
-                if tp == dtype.vector_3_float64:
-                    make_fake_coord = True
-                    ticks = {
-                        "formatter":
-                        lambda x: "(" + ",".join(
-                            [value_to_string(item, precision=2)
-                             for item in x]) + ")"
-                    }
-                elif tp == dtype.string:
-                    make_fake_coord = True
-                    ticks = {"formatter": lambda x: x}
-                if make_fake_coord:
-                    fake_unit = data_array.coords[dim].unit
-                    ticks["coord"] = data_array.coords[dim]
+                ticks = {
+                    "formatter":
+                    lambda x: "(" + ",".join(
+                        [value_to_string(item, precision=2)
+                         for item in x]) + ")"
+                }
+            elif tp == dtype.string:
+                make_fake_coord = True
+                ticks = {"formatter": lambda x: x}
             if make_fake_coord:
-                args = {"values": np.arange(self.shapes[name][dim])}
-                if fake_unit is not None:
-                    args["unit"] = fake_unit
-                var = Variable([dim], **args)
-            else:
-                var = data_array.coords[dim]
-        elif isinstance(axis, str):
-            # By convention, the last dim of the labels is the inner dimension,
-            # but note that for now two-dimensional labels are not supported in
-            # the plotting.
-            dim = data_array.labels[axis].dims[-1]
-            lab = axis
-            var = data_array.labels[lab]
+                fake_unit = data_array.coords[dim].unit
+                ticks["coord"] = data_array.coords[dim]
+        if make_fake_coord:
+            args = {"values": np.arange(self.shapes[name][dim])}
+            if fake_unit is not None:
+                args["unit"] = fake_unit
+            var = Variable([dim], **args)
         else:
-            raise RuntimeError("Unsupported axis found in 'axes': {}. This "
-                               "must be either a Scipp dimension "
-                               "or a string.".format(axis))
-        return dim, lab, var, ticks
+            var = data_array.coords[dim]
+        return dim, var, ticks
 
```

Comment:
Lots of commented code and seemingly unrelated changes in this file?

Reply:
Changes in this file were needed to get plotting with labels working again. I forgot about the commented code. Will remove